json/src/test/java/io/airlift/json/TestJsonCodec.java

```diff
@@ -257,4 +260,142 @@ public void testToJsonWithLengthLimitComplex()
+        // TODO MyRecordAdditionalGetter.getFoo is called and fails
+        assertThatThrownBy(() -> myRecordAdditionalGetterCodec.toJson(throwInGetFoo))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("io.airlift.json.TestJsonCodec$MyRecordAdditionalGetter could not be converted to JSON")
+                .rootCause()
+                .isInstanceOf(RuntimeException.class)
+                .hasMessage("getFoo throws now");
+
+        // TODO MyRecordAdditionalGetter.getFoo is called and returns a different value. Serialization cannot round-trip
+        assertEquals(
+                myRecordAdditionalGetterCodec.toJson(new MyRecordAdditionalGetter("my value", true, true, FakeGetterBehavior.DIFFERENT_VALUE)),
+                """
+                        {
+                          "additionalProperty" : "additional property value",
+                          "bar" : "there is no bar field in the record",
+                          "condition" : true,
+                          "foo" : "this is not a real getter for foo record component",
+                          "getFooBehavior" : "DIFFERENT_VALUE",
+                          "isCool" : true
+                        }""");
+    }
+
+    private static <T> void assertSerializationRoundTrip(JsonCodec<T> codec, T object, String expected)
+    {
+        assertEquals(codec.toJson(object), expected);
+        assertEquals(codec.fromJson(expected), object);
+    }
+
+    public record MyRecord(String foo) {}
+
+    public record MyRecordWithBeanLikeGetter(String foo)
+    {
+        public String getFoo()
+        {
+            return foo();
+        }
+    }
+
+    @JsonPropertyOrder(alphabetic = true)
+    public record MyRecordAdditionalGetter(
+            // a basic property
+            String foo,
+            // a boolean property that has additional getter and is-getter
+            boolean condition,
+            // a boolean property named as an is-getter
+            boolean isCool,
+            // technical property to drive the behavior during the test
+            FakeGetterBehavior getFooBehavior)
+    {
+        // Might shadow actual getter for foo -- the foo() method
+        public String getFoo()
+        {
+            return switch (getFooBehavior) {
+                case PASS -> foo();
+                case THROW -> throw new RuntimeException("getFoo throws now");
+                case DIFFERENT_VALUE -> "this is not a real getter for foo record component";
+            };
+        }
+
+        // Looks like a getter for "bar" property, there is no such record component
+        public String getBar()
+        {
+            return "there is no bar field in the record";
+        }
+
+        // Not a record component, but explicitly requested to be included in serialization
+        @JsonProperty
+        @SuppressWarnings("unused")
```

Comment:
Remove these suppressions

Reply:
i don't find it beneficial to have warnings reported by my IDE, even if IDE is wrong. i prefer to suppress them once rather than have them annoy me every time i work with the class

what is the benefit of removing the suppression?

Reply:
Because suppressions are a code smell. They are a signal that something is not quite right with the code. Additionally, they remove the warning but don't tell you when the warning no longer applies, so over time you end up with "dead" suppressions.

For this specific one, I would suggest you make your IDE ignore the unused warning when a method has  the `@JsonProperty` annotation. Since that expected to be an over-the-wire interface, there may be uses of the interface that aren't captured by code references.

Reply:
>  make your IDE ignore the unused warning when a method has the @JsonProperty annotation. Since that expected to be an over-the-wire interface

thanks for the tip!

that would make it harder to notice fields carried around (part of over-the-wire interface), but are otherwise unused

> Because suppressions are a code smell.

Agreed!
I use suppressions only when necessary to suppress warnings, so that i can still see warnings about real problems. Too many times i have seen my IDE warn me about a problem that's real and should be fixed, like lack of synchronization. 

I understand suppressions are even bigger code small then code warnings themselves, at least in this project. I will drop those annotations.